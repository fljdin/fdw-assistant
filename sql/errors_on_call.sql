-- "dummy" is a table that exist but is not used in any job
CREATE TABLE public.dummy (
    id serial primary key,
    name text not null
);

-- Should fail as the table does not exist
SELECT target, invocation FROM plan('{public.foo}');

-- Should return an empty set as the table has no configuration
SELECT target, invocation FROM plan('{public.dummy}');

-- copy(100) should fail because job_id does not exist
CALL copy(100);
