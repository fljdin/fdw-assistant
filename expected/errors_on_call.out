-- "dummy" is a table that exist but is not used in any job
CREATE TABLE public.dummy (
    id serial primary key,
    name text not null
);
CREATE TABLE
-- Should fail as the table does not exist
SELECT target, invocation FROM plan('{public.foo}');
ERROR:  relation "public.foo" does not exist
CONTEXT:  SQL function "plan" statement 1
-- Should return an empty set as the table has no configuration
SELECT target, invocation FROM plan('{public.dummy}');
 target | invocation 
--------+------------
(0 rows)

-- copy(100) should fail because job_id does not exist
CALL copy(100);
ERROR:  Job 100 not found
CONTEXT:  PL/pgSQL function copy(bigint) line 27 at RAISE
